execution/client/helpers.go

```diff
@@ -47,14 +47,14 @@ func processPayloadStatusResult(
 	result *engineprimitives.PayloadStatusV1,
 ) (*common.ExecutionHash, error) {
 	switch result.Status {
+	case engineprimitives.PayloadStatusValid:
+		return result.LatestValidHash, nil
 	case engineprimitives.PayloadStatusAccepted:
 		return nil, engineerrors.ErrAcceptedPayloadStatus
 	case engineprimitives.PayloadStatusSyncing:
 		return nil, engineerrors.ErrSyncingPayloadStatus
 	case engineprimitives.PayloadStatusInvalid:
-		return result.LatestValidHash, engineerrors.ErrInvalidPayloadStatus
-	case engineprimitives.PayloadStatusValid:
-		return result.LatestValidHash, nil
+		return nil, engineerrors.ErrInvalidPayloadStatus
```

Comment:
just noting: interesting that we were returning the LatestValidHash even if Invalid status was returned. Glad you’re getting rid of that and just returning nil

Reply:
100%